File: 0733-flood-fill/solution.java
class Solution {
    public int[][] floodFill(int[][] image, int sr, int sc, int color) {
        if(image[sr][sc] == color){
            return image;
        }

        int og = image[sr][sc];
        dfs(image,sr,sc,og,color);
        return image;
    }

    public void dfs(int[][] image, int i, int j, int og, int newColor){

        if(i < 0 || i >= image.length || j < 0 || j >= image[i].length || image[i][j] != og){
            return;
        }

        image[i][j] = newColor;
        dfs(image, i+1, j, og, newColor);
        dfs(image, i, j+1, og, newColor);
        dfs(image, i-1, j, og, newColor);
        dfs(image, i, j-1, og, newColor);
    }
}

File: 0733-flood-fill/solution.py
class Solution:
    def floodFill(self, image: List[List[int]], sr: int, sc: int, color: int) -> List[List[int]]:
        m = len(image)
        n = len(image[0])

        def dfs(image: List[list[int]], r: int, c: int, newcolor: int, oldcolor: int):
            if r < 0 or c < 0 or r >= m or c >= n or image[r][c] == newcolor or image[r][c] != oldcolor:
                return
            image[r][c] = newcolor

            dfs(image,r+1,c,newcolor,oldcolor)
            dfs(image,r,c+1,newcolor,oldcolor)
            dfs(image,r-1,c,newcolor,oldcolor)
            dfs(image,r,c-1,newcolor,oldcolor)
        
        dfs(image,sr,sc,color,image[sr][sc])
        return image
